packages/date/src/index.js

```diff
@@ -281,30 +260,52 @@ const formatMap = {
 	u: 'SSSSSS',
 	v: 'SSS',
 	// Timezone
-	e: 'zz',
+	/**
+	 * Return the timezone identifier for the given date.
+	 *
+	 * @param {Date} dateValue Date ISO string or object.
+	 *
+	 * @return {string} Formatted date.
+	 */
+	e( dateValue ) {
+		return formatTZ(
+			utcToZonedTime( dateValue, getActualTimezone() ),
+			'zzzz',
+			{ timeZone: getActualTimezone() }
+		);
+	},
 	/**
 	 * Gets whether the timezone is in DST currently.
 	 *
-	 * @param {Moment} momentDate Moment instance.
+	 * @param {Date} dateValue Date ISO string or object.
 	 *
 	 * @return {string} Formatted date.
 	 */
-	I( momentDate ) {
-		return momentDate.isDST() ? '1' : '0';
+	I() {
+		return ''; // @todo
+	},
+	O: 'xx',
+	P: 'xxx',
+	T( dateValue ) {
+		return formatTZ(
+			utcToZonedTime( dateValue, getActualTimezone() ),
+			'z',
+			{ timeZone: getActualTimezone() }
+		);
 	},
-	O: 'ZZ',
-	P: 'Z',
-	T: 'z',
 	/**
 	 * Gets the timezone offset in seconds.
 	 *
-	 * @param {Moment} momentDate Moment instance.
+	 * @param {Date|string} dateValue Date ISO string or object.
 	 *
 	 * @return {string} Formatted date.
 	 */
-	Z( momentDate ) {
+	Z( dateValue ) {
 		// Timezone offset in seconds.
-		const offset = momentDate.format( 'Z' );
+		const offset = dateFnsFormat(
+			utcToZonedTime( dateValue, 'UTC' ),
+			'XXX'
+		);
```

Comment:
Since this is going to be something like `+02:00` you want to cast `parts` to `Number`s.

Otherwise right now `parts` will be something like `[ 02, 00 ]`, which are interpreted as octals, making the calculation below incorrect.

Reply:
```js
const parts = offset.substring(1).split(':').map(Number);
```

should work

Reply:
Thanks! I've applied your suggestion.